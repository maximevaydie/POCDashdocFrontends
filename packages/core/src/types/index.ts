export * from "./company";
export * from "./availability";
export * from "./unavailability";
export * from "./bookingStatus";
export * from "./filters";
export * from "./misc";
export * from "./site";
export * from "./slot";
export * from "./statistics";
export * from "./zone";
export * from "./date";
export * from "./slotEvent";
export * from "./reporting";
