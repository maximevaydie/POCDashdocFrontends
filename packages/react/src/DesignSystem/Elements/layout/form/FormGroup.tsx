import React from "react";

import {Box} from "../Box";

export interface FormGroupProps {
    noCols?: boolean;
    className?: string;
    error?: any;
    htmlFor?: string;
    wideLabel?: boolean;
    label?: string;
    mandatory?: boolean;
    blueMandatory?: boolean;
    children: React.ReactNode;
    rightChildren?: React.ReactNode;
}

/**
 * @deprecated used on old forms
 */
export const FormGroup = (props: FormGroupProps) => {
    let formGroupClasses = `form-group ${props.className || ""} ${props.error ? "has-error" : ""}`;

    let errorHint;
    if (props.error) {
        errorHint = <span className="help-block">{props.error}</span>;
    }

    const labelProps = {
        className: `control-label ${!props.noCols && !props.wideLabel ? "col-sm-4" : ""} ${
            props.wideLabel ? "col-sm-5" : ""
        }`,
        htmlFor: props.htmlFor,
    };

    return (
        <Box className={formGroupClasses} mb={2}>
            {props.label && (
                <label {...labelProps}>
                    {props.label}
                    {(props.mandatory || props.blueMandatory) && (
                        <span>
                            {" "}
                            <b className={props.blueMandatory ? "text-info" : "text-danger"}>*</b>
                        </span>
                    )}
                </label>
            )}
            <div
                className={
                    props.label
                        ? `${props.wideLabel ? "col-sm-7" : ""} ${
                              !props.noCols && !props.wideLabel ? "col-sm-6" : ""
                          }`
                        : ""
                }
            >
                {props.children}
                {errorHint}
            </div>
            {props.rightChildren && (
                <div
                    style={{
                        lineHeight: "38px",
                        height: "0px", // Fix for margin bug in form-groups.
                    }}
                >
                    {props.rightChildren}
                </div>
            )}
        </Box>
    );
};
