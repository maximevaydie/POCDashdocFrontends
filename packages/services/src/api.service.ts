import {Api as API} from "dashdoc-utils";
import {Token} from "dashdoc-utils/dist/api/scopes/authentification";

import {Reports} from "./api-scopes/reportsApiScope";
import {authService} from "./auth.service";
import {storeService} from "./store.service";

/**
 * @guidedtour[epic=auth] Refresh triggered by the API
 * When the expiration date is reached, the API will refresh the token.
 */
function setToken(token: Token) {
    authService.setAuthToken(token);
}

/**
 * Boilerplate to give the token to the API from the redux store.
 */
function getToken(): Partial<Token> & {exp?: number} {
    const token = storeService.getState().auth.token;
    return token ?? {};
}

/**
 * @guidedtour[epic=auth] Logout triggered by the API
 * When the refresh token is invalid, the API will throw an error,
 * we decide to logout when a 401 happen.
 */
function onError(error: any) {
    if (error?.status === 401) {
        authService.logout();
    }
    // throw for the call to handle the error locally
    throw error;
}

const Api = new API({
    host: import.meta.env.VITE_API_HOST || window.location.origin,
    tokenPrefix: "JWT ",
    getToken,
    setToken,
    onError,
});

export const apiService = {
    ...Api,
    Reports: new Reports(Api.options),
};
