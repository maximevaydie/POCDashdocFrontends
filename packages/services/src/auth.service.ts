import {Logger} from "@dashdoc/web-core";
import {Token} from "dashdoc-utils/dist/api/scopes/authentification";
import decodeToken, {JwtPayload} from "jwt-decode";
import isEqual from "lodash.isequal";

import {resetReduxState} from "../redux/actions/reset";
import {loadAccount} from "../redux/reducers/accountReducer";
import {setToken} from "../redux/reducers/authReducer";
import {analyticsService} from "../services/analytics/analytics.service";
import {apiService} from "../services/api.service";
import {helpService} from "../services/help.service";
import {monitoringService} from "../services/monitoring.service";

import {storeService} from "./store.service";
import {tokenService} from "./token.service";

import type {PerishableToken} from "../types/auth";

function toPerishableToken(token: Token): PerishableToken | null {
    try {
        const {access, refresh} = token;
        if (access && refresh) {
            const payload = decodeToken<JwtPayload>(access);
            const exp = payload.exp;
            if (exp) {
                const result: PerishableToken = {access, refresh, exp};
                return result;
            }
        }
    } catch (e) {
        Logger.error("Error while converting token in perishableToken", e);
    }
    return null;
}

function setAuthToken(token: Token) {
    const perishableToken = toPerishableToken(token);
    if (perishableToken) {
        tokenService.persistToken(token);
        const dispatch = storeService.getDispatch();
        dispatch(setToken(perishableToken));
    } else {
        tokenService.persistToken(null);
    }
}

/**
 * @guidedtour[epic=auth] Login by form
 * User login from the standard form.
 */
async function loginByForm(data: {username: string; password: string}) {
    const token = await apiService.Auth.login({data});
    setAuthToken(token);
    const dispatch = storeService.getDispatch();
    dispatch(loadAccount() as any);
}

/**
 * @guidedtour[epic=auth] Login by code
 * From an temporary_auth_token.
 */
async function loginByCode(code: string | null) {
    if (code) {
        const token = await apiService.Auth.loginByCode({
            data: {code, type: "manager-code"},
        });
        setAuthToken(token);
        const dispatch = storeService.getDispatch();
        dispatch(loadAccount() as any);
    }
}

/**
 * @guidedtour[epic=auth] Login by moderation token
 * When a staff user want to log in a company account.
 */
async function loginByModerationToken(moderationToken: string | null) {
    if (moderationToken) {
        const token = await apiService.Auth.loginByModerationToken({
            data: {moderation_token: moderationToken},
        });
        setAuthToken(token);
        const dispatch = storeService.getDispatch();
        dispatch(loadAccount() as any);
    }
}

/**
 * @guidedtour[epic=auth] Another browser tab refresh the token
 * updates the token from the new local storage value.
 * Potentially, we could logout the user!
 */
async function localStorageSync() {
    const token = tokenService.getPersistedToken();
    const storeToken = storeService.getState().auth.token;
    if (!isEqual(token?.access, storeToken?.access)) {
        // The local storage changed!
        if (token) {
            setAuthToken(token);
        } else {
            authService.logout();
        }
    }
}

/**
 * @guidedtour[epic=auth] Logout.
 * We clear the local storage and reset the redux state!
 */
function logout() {
    tokenService.persistToken(null);
    //realtimeService.cleanup(); //TODO the realtime service is not yet generic enough to be used here
    monitoringService.cleanup();
    helpService.cleanup();
    analyticsService.cleanup();
    if (storeService.getState().auth.token?.access) {
        const dispatch = storeService.getDispatch();
        // Reset redux state to its initial state
        // (else we still have entities in state at next login
        // and they may be related to another account)
        dispatch(resetReduxState());
    }
}

function setPublicTokenForRestrictedDashdocAPIAccess(publicToken: string) {
    apiService.options.tokenPrefix = "PublicToken ";
    apiService.options.getToken = () => {
        return {access: publicToken, refresh: ""};
    };
}

export const authService = {
    setAuthToken,
    loginByForm,
    loginByCode,
    loginByModerationToken,
    logout,
    localStorageSync,
    setPublicTokenForRestrictedDashdocAPIAccess,
};
