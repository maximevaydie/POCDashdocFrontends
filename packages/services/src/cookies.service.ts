const getCookie = (name: string) => {
    if (document.cookie && document.cookie !== "") {
        var cookies = document.cookie.split(";");
        for (var i = 0; i < cookies.length; i++) {
            var cookie = cookies[i].trim();
            if (cookie.substring(0, name.length + 1) === name + "=") {
                return decodeURIComponent(cookie.substring(name.length + 1));
            }
        }
    }
    return null;
};

const setCookie = (name: string, value: string, days: number): null => {
    const date = new Date();
    date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000);
    document.cookie = name + "=" + value + "; expires=" + date.toUTCString() + "; path=/";
    return null;
};

const deleteCookie = (name: string): null => {
    document.cookie = name + "=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;";
    return null;
};

const setLanguageCookieAndReload = async (language: string) => {
    setCookie("django_language", language, 365);
    window.location.reload();
};

export const cookiesService = {getCookie, setCookie, deleteCookie, setLanguageCookieAndReload};
