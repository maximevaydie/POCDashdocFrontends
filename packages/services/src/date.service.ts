import {parseAndZoneDate} from "dashdoc-utils";
import {endOfDay, startOfDay} from "date-fns";

function sortByCreatedDeviceDate<T extends {created_device?: string | null; created?: string}>(
    collection: Array<T>,
    reverse = false
) {
    return sortByDate(collection, "created_device", "created", reverse);
}

function sortByDate<T, K extends keyof T>(
    collection: Array<T>,
    fieldName: K,
    backupFieldName: K,
    reverse = false
): Array<T> {
    let _true = -1,
        _false = 1;
    if (reverse) {
        _true = 1;
        _false = -1;
    }

    if (!backupFieldName) {
        backupFieldName = fieldName;
    }

    return collection.sort((a, b) => {
        const aValue = a[fieldName] ? a[fieldName] : a[backupFieldName];
        const bValue = b[fieldName] ? b[fieldName] : b[backupFieldName];
        if (!aValue) {
            return _true;
        } else if (!bValue) {
            return _false;
        }
        return new Date(aValue as any) < new Date(bValue as any) ? _true : _false;
    });
}

export function isDateInconsistent(
    startDate: Date | string | null,
    endDate: Date | string | null,
    timezone: string,
    minDateString?: string,
    maxDateString?: string
) {
    const minDate = minDateString ? parseAndZoneDate(minDateString, timezone) : null;
    const maxDate = maxDateString ? parseAndZoneDate(maxDateString, timezone) : null;
    return Boolean(
        startDate &&
            endDate &&
            ((maxDate && startDate > maxDate) || (minDate && endDate < minDate))
    );
}

export function getDatetimeRangeString(startDate: Date, endDate: Date) {
    const start = startOfDay(startDate).toISOString();
    const end = endOfDay(endDate).toISOString();
    const datetimeRange = [start, end].join(",");
    return datetimeRange;
}

export function reduceDates(
    dates: (string | undefined | null)[],
    comparison: "min" | "max"
): string | undefined {
    const nonEmptyDates = dates.filter((date) => date) as string[];
    const reducedDate = nonEmptyDates
        .map((date) => new Date(date))
        .reduce((acc: Date | undefined, date) => {
            if (!date) {
                return acc;
            } else if (!acc) {
                return date;
            } else if (comparison === "max") {
                return date > acc ? date : acc;
            } else {
                return date < acc ? date : acc;
            }
        }, undefined);

    return reducedDate?.toISOString();
}

export const dateService = {
    sortByDate,
    sortByCreatedDeviceDate,
    getDatetimeRangeString,
    reduceDates,
};
