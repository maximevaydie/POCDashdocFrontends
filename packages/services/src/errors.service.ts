import {Logger, t} from "@dashdoc/web-core";

function getStringValue(codeJson: any): string | null {
    if (typeof codeJson === "string") {
        return codeJson;
    }

    for (let key of Object.keys(codeJson)) {
        const value = codeJson[key];
        if (typeof value === "object") {
            return getStringValue(codeJson[key]);
        } else if (typeof value === "string") {
            return value;
        }
    }
    return null;
}

export function getMessageFromErrorCode(code: string): string | null {
    const ERROR_CODES: {[key: string]: string} = {
        origin_date_after_destination_date: t("errors.origin_date_after_destination_date"),
        trade_number_already_exists: t("errors.trade_number_already_exists"),
        remote_id_already_exists: t("errors.remote_id_already_exists"),
        max_length: t("errors.max_length"),
        blank: t("errors.field_cannot_be_empty"),
        multiple_rounds_different_load_categories: t(
            "errors.multiple_rounds_different_load_categories"
        ),
        user_already_manager_other_company: t("errors.user_already_manager_other_company"),
        user_already_manager_same_company: t("errors.user_already_manager_same_company"),
        duplicate_transport_template_name: t("transportTemplates.errorMessages.duplicateName"),
        duplicate_trucker_email: t("errors.duplicate_trucker_email"),
        unique_address: t("errors.duplicate_address"),
        cannot_update_carrier_ongoing_transport: t(
            "errors.cannot_update_carrier_ongoing_transport"
        ),
        cannot_mark_undone_wrong_role: t("errors.cannot_mark_undone_wrong_role"),
        cannot_mark_undone_transport_verified_or_amended: t(
            "errors.cannot_mark_undone_transport_verified_or_amended"
        ),
        cannot_mark_undone_activity_not_done: t("errors.cannot_mark_undone_activity_not_done"),
        cannot_mark_undone_next_activity_done: t("errors.cannot_mark_undone_next_activity_done"),
        cannot_mark_undone_break_resume_not_done: t(
            "errors.cannot_mark_undone_break_resume_not_done"
        ),
        started_trip_cannot_be_assigned: t("errors.started_trip_cannot_be_assigned"),
        cannot_cancel_on_site_wrong_role: t("errors.cannot_cancel_on_site_wrong_role"),
        cannot_cancel_on_site_transport_verified_or_amended: t(
            "errors.cannot_cancel_on_site_transport_verified_or_amended"
        ),
        cannot_update_multiple_rounds: t("errors.cannot_update_multiple_rounds"),
        slot_too_far_past: t("errors.slot_too_far_past"),
        slot_too_far_future: t("errors.slot_too_far_future"),
        cannot_make_company_debtor_of_draft_invoice_non_invoiceable: t(
            "errors.cannot_make_company_debtor_of_draft_invoice_non_invoiceable"
        ),
        company_does_not_exist_or_not_visible: t("errors.notFound"),
        cannot_credit_an_already_fully_credited_invoice: t(
            "errors.cannot_credit_an_already_fully_credited_invoice"
        ),
        invoice_should_be_at_least_finalized_to_be_credited: t(
            "errors.invoice_should_be_at_least_finalized_to_be_credited"
        ),
        cannot_credit_non_dashdoc_invoice: t("errors.cannot_credit_non_dashdoc_invoice"),
        invalid_invoice_uid: t("errors.invalid_invoice"),
        invalid_credit_note_uid: t("errors.invalid_credit_note"),
        cannot_finalize_non_draft_credit_note: t("errors.cannot_finalize_non_draft_credit_note"),
        invoicing_date_before_last_invoicing_date_outside_dashdoc: t(
            "errors.invoicing_date_before_last_invoicing_date_outside_dashdoc"
        ),
        invoicing_date_before_first_invoice: t("errors.invoicing_date_before_first_invoice"),
        cannot_mark_paid_non_finalized_credit_note: t(
            "errors.cannot_mark_paid_non_finalized_credit_note"
        ),
        cannot_mark_finalized_credit_note_as_not_paid: t(
            "errors.cannot_mark_finalized_credit_note_as_not_paid"
        ),
        invoicing_settings_mandatory: t("errors.invoicing_settings_mandatory"),
        negative_credit_note_total: t("errors.negative_credit_note_total"),
        email_in_use: t("errors.email_in_use"),
        email_already_exists_you_can_login: t("errors.email_already_exists_you_can_login"),
        password_too_short: t("errors.password_too_short"),
        password_too_similar: t("errors.password_too_similar"),
        password_too_common: t("errors.password_too_common"),
        password_entirely_numeric: t("common.passwordOnlyDigitsError"),
        invalid_opening_hours: t("errors.invalid_opening_hours"),
        slot_not_available: t("errors.slot_not_available"),
        not_within_booking_window: t("errors.slot_start_not_within_booking_window"),
        slot_not_in_the_future: t("errors.slot_not_in_the_future"),
        notice_period_error: t("errors.slot_notice_period_error"),
        security_protocol_not_accepted: t("common.securityProtocol.mandatory"),
        missing_custom_field: t("common.slot_custom_fields_out_of_sync"),
        invalid_custom_fields_json: t("common.invalid_format"),
        missing_active_subscription: t("errors.waste_shipment.missing_active_subscription"),
        creator_must_be_waste_actor: t("errors.waste_shipment.creator_must_be_waste_actor"),
        big_query_unavailable_in_this_env: t("errors.big_query_unavailable_in_this_env"),
        invalid_slot_transition_timestamp: t("errors.invalid_slot_transition_timestamp"),
        non_invoiceable_customer_to_invoice: t("errors.non_invoiceable_customer_to_invoice"),
        invalid_customer_to_invoice: t("errors.invalid_customer_to_invoice"),
        total_weight_must_be_equal_to_one: t("errors.total_weight_must_be_equal_to_one"),
        missing_coordinates: t("errors.optimization.missing_coordinates"),
        missing_deliveries: t("errors.optimization.missing_deliveries"),
        missing_loads: t("errors.optimization.missing_loads"),
        missing_linear_meters: t("errors.optimization.missing_linear_meters"),
        overload: t("errors.optimization.overloadWithoutParameters"),
        missing_distance: t("errors.optimization.missing_distance"),
        no_solution_found: t("errors.optimization.no_solution_found"),
    };
    const errorMessage = ERROR_CODES[code];
    if (errorMessage !== undefined) {
        return errorMessage;
    }
    return null;
}

function getMessageFromError(errorJson: any): string | null {
    /* Error JSON usually look something like:
     * {
     *     remote_id: {
     *         detail: "There is already a transport with the remote_id 12345",
     *         code: "invalid",
     *     }
     * }
     */

    // First we try to find a specific error code for which we have
    // a user friendly error message.
    if (errorJson.code) {
        const code = getStringValue(errorJson.code);
        // @ts-ignore
        const message = getMessageFromErrorCode(code);
        if (message !== null) {
            return message;
        }
    }

    // If that's not the case, then we try to find an error message
    // directly from the server.
    if (errorJson.detail) {
        const message = getStringValue(errorJson.detail);
        if (message !== null) {
            return message;
        }
    }

    // We iterate other the different fields with an error and stop at soon as
    // we find a error message
    for (let key of Object.keys(errorJson)) {
        const message = getMessageFromError(errorJson[key]);
        if (message !== null) {
            return message;
        }
    }

    return null;
}

/**
 * MADE WITH ERRORS RETURNED FROM POST ON /api/companies/
 * WITH FORMIK ERROR HANDLING IN MIND
 * PLEASE CORRECT THIS FUNCTION IF NEEDED
 */
export async function getErrorMessagesFromServerError(responseError: any) {
    function recursiveErrorMessageGetter(
        responseError: any,
        errorMessages: any = {},
        parentKey = ""
    ) {
        for (let key of Object.keys(responseError)) {
            if (responseError[key].code) {
                if (!Array.isArray(responseError[key].code)) {
                    errorMessages[`${parentKey}${key}`] =
                        getMessageFromErrorCode(responseError[key].code) ||
                        responseError[key].detail;
                } else {
                    // Case with multiple errors for one field
                    const messages: string[] = [];
                    for (const code of responseError[key].code) {
                        const stringCode = getStringValue(code);
                        if (stringCode !== null) {
                            const message = getMessageFromErrorCode(stringCode);
                            if (message !== null) {
                                messages.push(message);
                            }
                        }
                    }
                    if (messages.length > 0) {
                        errorMessages[`${parentKey}${key}`] = messages.join("\n");
                    } else {
                        errorMessages[`${parentKey}${key}`] = Array.isArray(
                            responseError[key].detail
                        )
                            ? responseError[key].detail.join("\n")
                            : responseError[key].detail;
                    }
                }
            } else {
                recursiveErrorMessageGetter(
                    responseError[key],
                    errorMessages,
                    `${parentKey}${key}.`
                );
            }
        }
        return errorMessages;
    }
    try {
        const responseJson = (responseError.json && (await responseError.json())) || responseError;
        return recursiveErrorMessageGetter(responseJson);
    } catch (err) {
        Logger.warn(err);
        return t("common.error");
    }
}

export async function getErrorMessageFromServerError(
    responseError: any,
    genericErrorMessage: string = t("common.error")
) {
    try {
        const clone = responseError.clone();
        const errorJson = await clone.json();
        const errorMessage = getMessageFromError(errorJson);
        if (errorMessage) {
            return errorMessage;
        }
    } catch (err) {
        Logger.warn(err);
    }

    // if we can't find a specific error message, we return a generic one
    switch (responseError.status) {
        case 403: {
            return t("errors.forbidden");
        }
        case 404: {
            return t("errors.notFound");
        }
        default: {
            return genericErrorMessage;
        }
    }
}
