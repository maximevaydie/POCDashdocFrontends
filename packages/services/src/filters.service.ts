import {addDays, endOfDay, startOfDay, tz} from "services/date";
import {RequestFilters, TzDate} from "types";

function create(dateFrom: TzDate, siteId: number, dayRange?: number): RequestFilters {
    const {start, end} = getStartAndEnd(dateFrom, dayRange);
    const result: RequestFilters = {
        site: siteId,
        start,
        end,
    };
    return result;
}

function getStartAndEnd(
    dateFrom: TzDate,
    dayRange?: number
): {end: string; endDate: TzDate; start: string; startDate: TzDate} {
    // We want to start from the beginning of the day
    const dateFromBegin = startOfDay(dateFrom);

    const siteDatePlusRange =
        dayRange !== undefined ? addDays(dateFromBegin, Math.max(0, dayRange - 1)) : dateFromBegin;
    const endOfRange = endOfDay(siteDatePlusRange);

    const timezone = dateFromBegin.timezone;
    const start = tz.dateToISO(dateFromBegin);
    const startDate = tz.convert(start, timezone);

    const end = tz.dateToISO(endOfRange);
    const endDate = tz.convert(end, timezone);
    return {
        start,
        startDate,
        end,
        endDate,
    };
}

export const filtersService = {
    create,
    getStartAndEnd,
};
