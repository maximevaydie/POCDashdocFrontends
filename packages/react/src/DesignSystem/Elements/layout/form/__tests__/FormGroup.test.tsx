import {itRendersCorrectlyWith} from "@dashdoc/web-core/src/testing";
import React from "react";

import {FormGroup} from "../FormGroup";

describe("FormGroup", () => {
    itRendersCorrectlyWith(
        "children",
        <FormGroup>
            <input type="text" className="form-control" />
        </FormGroup>
    );

    itRendersCorrectlyWith(
        "error",
        <FormGroup error="this is a test error">
            <input type="text" className="form-control" />
        </FormGroup>
    );

    itRendersCorrectlyWith(
        "label",
        <FormGroup label="This is a helpful label">
            <input type="text" className="form-control" />
        </FormGroup>
    );

    itRendersCorrectlyWith(
        "no cols",
        <FormGroup noCols={true}>
            <input type="text" className="form-control" />
        </FormGroup>
    );
});
