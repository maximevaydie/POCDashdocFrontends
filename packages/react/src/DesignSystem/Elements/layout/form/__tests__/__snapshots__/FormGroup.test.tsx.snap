// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGroup renders correctly with children 1`] = `
<div
  className="form-group   css-1f978ju"
>
  <div
    className=""
  >
    <input
      className="form-control"
      type="text"
    />
  </div>
</div>
`;

exports[`FormGroup renders correctly with error 1`] = `
<div
  className="form-group  has-error css-1f978ju"
>
  <div
    className=""
  >
    <input
      className="form-control"
      type="text"
    />
    <span
      className="help-block"
    >
      this is a test error
    </span>
  </div>
</div>
`;

exports[`FormGroup renders correctly with label 1`] = `
<div
  className="form-group   css-1f978ju"
>
  <label
    className="control-label col-sm-4 "
  >
    This is a helpful label
  </label>
  <div
    className=" col-sm-6"
  >
    <input
      className="form-control"
      type="text"
    />
  </div>
</div>
`;

exports[`FormGroup renders correctly with no cols 1`] = `
<div
  className="form-group   css-1f978ju"
>
  <div
    className=""
  >
    <input
      className="form-control"
      type="text"
    />
  </div>
</div>
`;
